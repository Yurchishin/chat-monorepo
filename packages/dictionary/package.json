{
  "name": "@monorepo/dictionary",
  "version": "0.0.1",
  "description": "",
  "keywords": [],
  "author": "Yurii Rybak <yurarary@gmail.com>",
  "license": "ISC",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "tsc -p tsconfig.json",
    "watch": "tsc -p tsconfig.json --watch",
    "start": "node dist/index.js",
    "start:build": "npm run build & npm run start",
    "lint": "npm run lint:eslint npm run lint:tslint",
    "lint:fix": "npm run lint:eslint-fix npm run lint:tslint-fix",
    "lint:eslint": "eslint --ext .ts src",
    "lint:eslint-fix": "eslint --fix --ext .ts src",
    "lint:tslint": "tslint -c ./tslint.json -p ./tsconfig.json",
    "lint:tslint-fix": "tslint -c ./tslint.json -p ./tsconfig.json --fix"
  },
  "devDependencies": {
    "typescript": "^3.7.5"
  },
  "dependencies": {
    "@types/node": "^13.1.8"
  }
}
