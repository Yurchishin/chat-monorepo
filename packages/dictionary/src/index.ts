export * from './Hosts'
export * from './NodeEnv'
export const Hui = 'hi'
