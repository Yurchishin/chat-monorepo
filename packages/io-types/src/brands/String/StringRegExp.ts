import * as t from 'io-ts'

export interface IStringRegExpBrand {
    readonly StringRegExp: unique symbol
}

export const stringRegExp = (regExp: RegExp) => t.brand(
    t.string,
    (n): n is t.Branded<string, IStringRegExpBrand> => regExp.test(n),
    'StringRegExp',
)
