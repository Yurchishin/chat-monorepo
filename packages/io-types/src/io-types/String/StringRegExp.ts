import * as t from 'io-ts'
import * as E from 'fp-ts/lib/Either'

export interface IStringRegExpC extends t.Type<string> {}

export const stringRegExp = (regExp: RegExp): IStringRegExpC => new t.Type<string>(
    'StringRegExp',
    t.string.is,
    (u, c) =>
        E.either.chain(
            t.string.validate(u, c),
            s => regExp.test(s) ? t.success(s) : t.failure(u, c),
        ),
    t.identity,
)
