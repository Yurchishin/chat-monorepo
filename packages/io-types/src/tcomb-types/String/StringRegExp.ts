import t from 'tcomb-validation'
import R from '@chat/ramda'

export const StringRegExp = (regExp: RegExp) => t.irreducible(
    'StringRegExp',
    R.test(regExp),
)
