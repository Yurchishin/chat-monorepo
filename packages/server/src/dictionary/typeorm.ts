export const TYPEORM_CONNECTION = 'postgres'
export const TYPEORM_ENTITIES = 'dist/entity/**/*.js'
export const TYPEORM_MIGRATIONS = 'dist/migration/**/*.js'
export const TYPEORM_ENTITIES_DIR = 'src/entity'
export const TYPEORM_MIGRATIONS_DIR = 'src/migration'

