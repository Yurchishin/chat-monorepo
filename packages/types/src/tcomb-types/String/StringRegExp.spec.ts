import {UserPasswordRegExp} from "@chat/dictionary";
import {StringRegExp} from './StringRegExp'
import {expectTypeSuccess, expectTypeFailure} from "../../tests"

describe('StringRegExp(UserPasswordRegExp)', () => {
	const regExp = StringRegExp(new RegExp(UserPasswordRegExp))

	expectTypeSuccess(regExp, 'OneTwo123***')
	expectTypeFailure(regExp, 'onetwo123***')
	expectTypeFailure(regExp, 'OneTwo123')
	expectTypeFailure(regExp, '55555')
	expectTypeFailure(regExp, 1)
})
