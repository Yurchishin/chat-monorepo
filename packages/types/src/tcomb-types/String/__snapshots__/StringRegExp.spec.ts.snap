// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringRegExp(UserPasswordRegExp) expectTypeFailure(1): TYPE_FAILURE::StringRegExp(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Invalid value 1 supplied to StringRegExp",
      "path": Array [],
    },
  ],
  "value": 1,
}
`;

exports[`StringRegExp(UserPasswordRegExp) expectTypeFailure(55555): TYPE_FAILURE::StringRegExp(55555) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "55555",
      "expected": [Function],
      "message": "Invalid value \\"55555\\" supplied to StringRegExp",
      "path": Array [],
    },
  ],
  "value": "55555",
}
`;

exports[`StringRegExp(UserPasswordRegExp) expectTypeFailure(OneTwo123): TYPE_FAILURE::StringRegExp(OneTwo123) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "OneTwo123",
      "expected": [Function],
      "message": "Invalid value \\"OneTwo123\\" supplied to StringRegExp",
      "path": Array [],
    },
  ],
  "value": "OneTwo123",
}
`;

exports[`StringRegExp(UserPasswordRegExp) expectTypeFailure(onetwo123***): TYPE_FAILURE::StringRegExp(onetwo123***) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "onetwo123***",
      "expected": [Function],
      "message": "Invalid value \\"onetwo123***\\" supplied to StringRegExp",
      "path": Array [],
    },
  ],
  "value": "onetwo123***",
}
`;
